### ISSUE TITLE
[Wave Collect][QBO] Accounting - QBO menu only appears after revisiting Accounting after synchronization

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#40513]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.63-0
**Reproducible in staging?:** y
**Reproducible in production?:** No,  Newfeature
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Launch New Expensify app.
2. Go to Account settings > Profile > Workspaces.
3. Select any workspace.
4. Go to Accounting tab.
5. Set up with a QBO account.
6. Complete the setup flow.
## Expected Result:
After synchronizing on ND, it will show Import, Export and Advanced options.

## Actual Result:
After synchronizing on ND, it still shows Set up button.
Only after going to a different tab and returning to Accounting page, Import, Export and Advanced options appear.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/be58b31b-5110-462a-9985-b8359e23aee7

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by hayata-suenaga** [comment]
Removed the deploy blocker label as this is still behind beta

**Comment by aldo-expensify** [comment]
@hayata-suenaga this is the same you are working on here right? https://github.com/Expensify/App/pull/40375

**Comment by hayata-suenaga** [comment]
I don't think the PR you linked is going to solve the issue. The issue is that the pusher update sent from the backend might not be property reflected in the UI.

**Comment by hayata-suenaga** [comment]
@aldo-expensify, when the sync is finished, the connections data is saved and that is going to send a pusher notification, right?

**Comment by aldo-expensify** [comment]
> I don't think the PR you linked is going to solve the issue. The issue is that the pusher update sent from the backend might not be property reflected in the UI.

I saw the video now and you are right, I thought this was the problem of the infinite spinner of the policy connections HOC

**Comment by aldo-expensify** [comment]
> @aldo-expensify, when the sync is finished, the connections data is saved and that is going to send a pusher notification, right?

yes, from the video, it looks like the final update with the connection data was missed but the updates with the sync progress from the IS did arrive.

**Comment by hayata-suenaga** [comment]
@aldo-expensify, can I assign you to this issue so that you can investigate more why the last pusher update is not received or reflected on the UI?

**Comment by aldo-expensify** [comment]
@hayata-suenaga it is a good debugging exercise 😆 

@kavimuru do you know which account email was used in that video? having the email should help with finding the logs and then we can check if there was an error when sending the pusher updates

**Comment by CortneyOfstad** [comment]
Bump @kavimuru on the comment above — thanks!

**Comment by hayata-suenaga** [comment]
> @hayata-suenaga it is a good debugging exercise for you 😆
haha 😆 I might be able to work on it, but I'm tied with reviewing the several QBO related PRs and with opening a few PRs myself to fix QBO related issues. thought you'd be better equipped with figuring out this as you implemented the pusher updates in the integration server 😄

**Comment by hayata-suenaga** [comment]
QA team will respond quickly if [you message them in Slack](https://expensify.slack.com/archives/C9YU7BX5M/p1713981675994619) 😄

**Comment by hayata-suenaga** [comment]
The tester @kavimuru [is currently offline](https://expensify.slack.com/archives/C9YU7BX5M/p1713988811914819?thread_ts=1713981675.994619&cid=C9YU7BX5M). They gonna update once they come back online

**Comment by kbecciv** [comment]
@hayata-suenaga The tester doesn't remember the account email, but they were able to reproduce the issue with a different one.
Account email: [applausetester+kh2404006@applause.expensifail.com](mailto:applausetester+kh2404006@applause.expensifail.com)

https://github.com/Expensify/App/assets/93399543/6ad40d75-6fea-4282-84ff-497c864044d0

**Comment by trjExpensify** [comment]
This has been updating for me without navigating away or needing a refresh on the latest staging.

**Comment by aldo-expensify** [comment]
> This has been updating for me without navigating away or needing a refresh on the latest staging.

I has not happened for me either, but given that the connection appears after navigating away and coming back, I think it is because the pusher event with the policy connection data didn't reach the client for some reason.

**Comment by trjExpensify** [comment]
Can we do anything about or is it an isolated instance of a flaky pusher event?

**Comment by CortneyOfstad** [comment]
@aldo-expensify bump on TRJ's question above ^^^ Thanks!

**Comment by aldo-expensify** [comment]
> is it an isolated instance of a flaky pusher event?

I think we may have two cases:

- Flaky pusher event makes us miss the update on the policy with the connections (less likely)
- You are already logged in in OldDot with a different account (more likely if you are a tester)

**Comment by aldo-expensify** [comment]
Here are some logs for the second case: 

https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2287c1303fcdf82dbf-YVR%22)+AND+timestamp:[2024-04-29T16:59:53.748Z+TO+2024-04-29T18:59:53.748Z]&index=logs_expensify-028086

There is an alternation between the logged email being aldo+qbo-test23@expensifail.com and aldo+qbo-test24@expensifail.com, which probably causes bugs

**Comment by CortneyOfstad** [comment]
@hayata-suenaga do you have any thoughts on @aldo-expensify's logs above?

**Comment by hayata-suenaga** [comment]
this also haven't happened to me. I think we can close the issue or change the priority to weekly and wait for Applause testing cycle to confirm this wont't happen

**Comment by trjExpensify** [comment]
Interestingly, this happened to me on the adhoc build earlier today. Not
sure if staging is better, but maybe we do have an intermittent problem
here.

**Comment by hayata-suenaga** [comment]
I have been testing in different PRs and also staging, but hasn't encountered this issue for a while.
cc: @aldo-expensify for keeping you in the loop

**Comment by trjExpensify** [comment]
This is good for me on the latest staging. So let's close it out!

**Comment by izarutskaya** [comment]
This issue is still reproducible 
Build v1.4.79-2

https://github.com/Expensify/App/assets/115492554/d68ea449-bce6-4837-baba-d8fd6c46d441


https://github.com/Expensify/App/assets/115492554/11a29ace-1210-4a09-b787-91916185bf6c

**Comment by hayata-suenaga** [comment]
probably same reason as [this one](https://github.com/Expensify/App/issues/40513#issuecomment-2083330770)

what do you think @aldo-expensify 🤔

**Comment by aldo-expensify** [comment]
Hmm, I found some logs I think:

- When `ConnectPolicyToQuickbooksOnline ` is called: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2288edb4e2a85082e8-GIG%22)+AND+timestamp:[2024-06-05T04:19:14.475Z+TO+2024-06-05T06:19:14.475Z]&index=logs_expensify-028778 . It is strange that the Authtoken doesn't seem to be set properly, the email shows as: `we@dont.know`. Not sure if this matters.
- Logs for the IS doing the first connection: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%22DfiTPH%22)%20AND%20timestamp%3A%5B2024-06-05T04%3A21%3A32.922Z%20TO%202024-06-05T06%3A21%3A32.922Z%5D&index=_all . There is an authtoken permission error: `Insufficient privileges for token type policyAutoSync and method SendIntegrationProgressUpdate`. Seems like `SendIntegrationProgressUpdate` can't be called with `policyAutoSync` token. cc @yuwenmemon . I don't think if this error is related to the problem seen [here](https://github.com/Expensify/App/issues/40513#issuecomment-2149480064).
- Logs for the quickbooks/oauth_callback.php: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%2288edb83d18fc82eb-GIG%22)%20AND%20timestamp%3A%5B2024-06-05T04%3A21%3A31.806Z%20TO%202024-06-05T06%3A21%3A31.806Z%5D&index=_all . I don't see problems here.

In summary, I don't think we are in the case where there is an existing session in old dot with a different email because if that was the case, we would see a mix of emails in the logs and I see `applausetester+888892@applause.expensifail.com`. That leaves us with:

- They are missing pusher events, or
- The error thrown by `SendIntegrationProgressUpdate` is causing problems

**Comment by hayata-suenaga** [comment]
[Asking the QA team for retest with additional instructions](https://expensify.slack.com/archives/C9YU7BX5M/p1717692237722309)

**Comment by trjExpensify** [comment]
Fresh repro from me: 

https://github.com/Expensify/App/assets/16232057/404b7b40-50d2-41b8-b8e1-349fd95312f5

**Comment by yuwenmemon** [comment]
Oh nice, I'll fix that one for the authToken. But `SendIntegrationProgressUpdate` not working is not going to be the source of our problem here. 

My bet is that there's a missing OnyxUpdate somewhere.

**Comment by aldo-expensify** [comment]
> My bet is that there's a missing OnyxUpdate somewhere.

Seems like this is the cause really: https://expensify.slack.com/archives/C036QM0SLJK/p1717705238059739?thread_ts=1717422206.719739&cid=C036QM0SLJK

**Comment by kbecciv** [comment]
Issue is reproducible after trying the following methods posted here - https://expensify.slack.com/archives/C9YU7BX5M/p1717692237722309

https://github.com/Expensify/App/assets/93399543/4fd6e282-f7bb-4865-bc2c-e5e1c0226cd4

**Comment by hayata-suenaga** [comment]
The reproduction step in the above video is not correct as Aldo suggested [here](https://expensify.slack.com/archives/C9YU7BX5M/p1718038904389259?thread_ts=1717692237.722309&cid=C9YU7BX5M), but the cause of the issue was determined to be an issue with pushers.

**Comment by hayata-suenaga** [comment]
The pusher issue should have been solved by [this PR](https://github.com/Expensify/Web-Expensify/pull/42292)

Closing the issue
